nan_maybe_43_inl.h

```diff
@@ -109,8 +109,22 @@ inline Maybe<bool> DefineOwnProperty(
   , v8::PropertyAttribute attribs = v8::None) {
   v8::Isolate *isolate = v8::Isolate::GetCurrent();
   v8::HandleScope scope(isolate);
+#if NODE_MODULE_VERSION >= NODE_4_0_MODULE_VERSION
   return obj->DefineOwnProperty(isolate->GetCurrentContext(), key, value,
                                 attribs);
+#else
+  Maybe<v8::PropertyAttribute> maybeCurrent =
+      obj->GetPropertyAttributes(isolate->GetCurrentContext(), key);
+  if (maybeCurrent.IsNothing()) {
+    return Nothing<bool>();
+  }
+  v8::PropertyAttribute current = maybeCurrent.FromJust();
+  return !(current & v8::DontDelete) ||                     // configurable OR
+                  (!(current & v8::ReadOnly) &&             // writable AND
+                   !((attribs ^ current) & ~v8::ReadOnly))  // same excluding RO
+             ? Just<bool>(obj->ForceSet(key, value, attribs))
+             : Nothing<bool>();
+#endif
 }
```

Comment:
I realize that it is by design, but this code is identical to the block in the `nan_maybe_pre_43_inl.h` version: https://github.com/nodejs/nan/blob/master/nan_maybe_pre_43_inl.h#L151

How about we add a new file called, `nan_maybe.h` or `nan_maybe_inl.h` and move the logic of deciding which header to use from `nan.h` to this new file, along with a block of common code (such as this duplicated block)

I'd hate to see future maintenance improve one block and possibly forget to include the other.

Reply:
I thought of that while writing it, but it is not fully identical. There is some `Maybe`-stuff going on before the `return` statement, which makes deduplication harder.

Reply:
...but not impossible.   Meanwhile, my desire for deduplication shouldn't hold up this PR.  We can always take a stab at that later.